web/packages/teleport/src/Integrations/status/AwsOidc/Tasks/Tasks.tsx

```diff
@@ -71,14 +90,21 @@ export function Tasks() {
     return <Danger>{serverSidePagination.attempt.statusText}</Danger>;
   }
 
-  function close(resolved: boolean) {
+  function closeTask(resolved: boolean) {
     if (resolved) {
-      serverSidePagination.modifyFetchedData(p => ({
-        ...p,
-        agents: p.agents.filter(t => t.name !== showTask.name),
-      }));
+      // If there are multiple pages, we would rather refresh the table with X results rather than
+      // use modifyFetchedData to remove the item.
+      serverSidePagination.fetch();
+    }
+    history.replace(history.location.pathname);
+  }
+
+  function openTask(task: UserTask) {
+    if (selectedTask == undefined) {
+      const urlParams = new URLSearchParams();
+      urlParams.append('task', task.name);
+      history.replace(`${history.location.pathname}?${urlParams.toString()}`);
```

Comment:
is the ideal result here to "do nothing" if there is a selected task? if so, i think itd be better to remove indentation by reversing this conditional. like this
```suggestion
  function openTask(task: UserTask) {
    if (selectedTask != undefined) {
      return
    }
    const urlParams = new URLSearchParams();
    urlParams.append('task', task.name);
    history.replace(`${history.location.pathname}?${urlParams.toString()}`);
```